### PR TITLE
GEP: Add Timeouts GEP first cut

### PR DESCRIPTION
**Pull request by youngnick**
**What type of PR is this?**
/kind gep

**What this PR does / why we need it**:
This PR adds a first attempt at systematizing the discussion around timeouts in Gateway API.

It also adds support for Markdown tables and MermaidJS diagrams to our mkdocs installation.

**Which issue(s) this PR fixes**:
Updates #1742

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by youngnick** [comment]
I'd recommend checking the preview at https://deploy-preview-1744--kubernetes-sigs-gateway-api.netlify.app/geps/gep-1742/ to see the diagrams properly.

File: site-src/geps/gep-1742.md
Comment (dprotaso) [inline site-src/geps/gep-1742.md:16]: It would be good to capture what types of timeouts are relevant to a broad set of users even if implementations currently don't support it.

Then we have enough data to say to implementors - 'if you add support for X,Y timeouts you'll handle 80% of use cases'

File: site-src/geps/gep-1742.md
Comment (keithmattix) [inline site-src/geps/gep-1742.md:20]: Support for all possible timouts seems like a good non-goal

File: site-src/geps/gep-1742.md
Comment (keithmattix) [inline site-src/geps/gep-1742.md:13]: This is where we have to decide whether or not we want to be opinionated about what a timeout is

**Review by keithmattix — APPROVED** [review]

File: site-src/geps/gep-1742.md
Comment (Xunzhuo) [inline site-src/geps/gep-1742.md:27]: It is happy to see this moves on.

File: site-src/geps/gep-1742.md
Comment (Xunzhuo) [inline site-src/geps/gep-1742.md:30]: Yes, we should focus on L7 firstly.

**Review by Xunzhuo — COMMENTED** [review]
Thanks for ping me to take a review @shaneutt. Happy to see this move on @youngnick.

Initial content looks good to me.

**Review by mlavacca — APPROVED** [review]
Thanks for asking me for the review. This document seems totally reasonable as a first iteration and a good starting point 👍

**Review by shaneutt — APPROVED** [review]
I like that for now we're just stating what we'd like to do and providing some groundwork information about the relevant proxies at play so we can continue to build from here (the mermaid diagrams are nice too). I think as provisional, this is a good place to start the conversation and I have no blockers. :+1:

**Review by Xunzhuo — APPROVED** [review]
looks good for me, and there is no blocker from me, maybe need another review from @robscott before getting merged this.

**Comment by youngnick** [comment]
This GEP definitely needs some more information to be viable, but could be merged as-is because it's provisional anyway.

**Review by Xunzhuo — COMMENTED** [review]
/lgtm 

Thanks @youngnick !

**Comment by youngnick** [comment]
oh, actually, we already have another maintainer review, sorry @Xunzhuo.

/unhold

**Pull request merged**
